woocommerce/woocommerce-android
# Issue/11359 unavailable cards

Part of: #11359 


⚠️ Don't merge until https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2992 is merged and FluxC version updated.

### Description
This PR includes the following changes:
1. Update the DashboardWidget model to include an availability flag (we might need to improve this to support different statuses, and not just availability, but this can be done in a separate PR)
2. Re-work the `DragAndDropItemsList`:
    - Make the component more generic and accepting custom UIs for the items.
    - Allow marking some items as non-draggable.
3. Update the Widget settings screen to handle unavailable widgets.
4. Mark the `Performance` and `Top Performers` widgets as unavailable when the site doesn't have orders.

Reviewing bc5845cb39b1f8388cef4da66d8a8edf02b26d88 (about the `DragAndDropItemsList`) separately would make the review of other parts easier.

I'll handle the other cards (Blaze and the Onboarding) in separate PRs.

### Testing instructions
1. Create a new Jurassic Ninja site.
2. Open the app and sign in to the site.
3. Open the dashboard screen and confirm the Performance and Top Performers cards are hidden.
4. Open the widget editor screen and confirm both cards are marked as unavailable.
5. Test drag and dropping, and confirm it works as expected.
6. Create some orders.
7. Confirm the cards are shown.

### Images/gif
